### DISCUSSION TITLE
Augmenty for spacy training pipe config.cfg

### DISCUSSION THREAD

**Discussion by leonballonigomes** [KennethEnevoldsen/augmenty#117] [Q&A]
Good afternoon,
I`ve spent sometime studying the package and I would like to know if it`s possible to use it with the config.cfg file proposed by Spacy https://spacy.io/usage/training. 
If yes, could you send me some info about how to do it, please?

Thank you.

**Comment by KennethEnevoldsen** [comment]
Indeed it is!

I have an example [here](https://github.com/centre-for-humanities-computing/DaCy/blob/c775e294401101460d2938a7904116fe2354ec74/training/v0.1.1/configs/config_large.cfg#L127) where I use it to train DaCy (a danish pipeline for spacy):
```
[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths:train}
max_length = 500
gold_preproc = false
limit = 0

[corpora.train.augmenter]
@augmenters = "dacy_augmenter.v1"
```

Where `dacy_augmenter.v1` can be any augmenter in augmenty (mine is a custom one I made in this [script](https://github.com/centre-for-humanities-computing/DaCy/blob/main/training/v0.1.1/danish_augmenter.py)). 

When calling spacy train using your new config.cfg it will not know that the augmenty augmenters exist. The way to resolve this is to use the `--code` argument and pass it a python file (let us call it `augmenty_augmenters.py`) this file can look like so:

```
import augmenty
# will add all augmenters to spacy catalogue
```

So you final command will look something like this:
`spacy train [...] --code augmenty_augmenters.py`  

But you might want to use multiple of the augmenters and can easily combine these (see [here](https://kennethenevoldsen.github.io/augmenty/tutorials/introduction.html#Combining-augmenters) for an example). You can then add  your newly constructed augmenter using the spacy decorator: `@spacy.registry.augmenters("my_cool_augmenter.v1")`

I also wrote an example of this in this [issue](https://github.com/KennethEnevoldsen/augmenty/issues/78) 😄

**Reply by leonballonigomes** [reply]
Thank you! I'll try this.
